issue_summarization_github_issues/setup_a_kubeflow_cluster.md

```diff
@@ -0,0 +1,174 @@
+# Setup Kubeflow
+
+In this part, you will setup kubeflow on an existing kubernetes cluster.
+
+## Requirements
+
+*   A kubernetes cluster
+*   `kubectl` CLI pointing to the kubernetes cluster
+    *   Make sure that you can run `kubectl get nodes` from your terminal
+        successfully
+*   The ksonnet CLI: [ks](https://ksonnet.io/#get-started)
+*   `gcloud` CLI if using GCP
+
+## Create a kubernetes namespace
+
+We will do all our work in a new kubernetes namespace so that it doesn't overlap
+with any existing kubernetes resources.
+
+```
+KUBE_NS=issue-summarization-model
+kubectl create namespace ${KUBE_NS}
+```
+
+## Deploy Kubeflow
+
+We will be using ksonnet(ks) to deploy kubeflow into your cluster.
+
+Initialize a directory to contain your ksonnet application. This directory will
+contain all the kubernetes manifests used to deploy kubeflow.
+
+```
+APP_NAME=kubeflow-app
+ks init ${APP_NAME}
+```
+
+Install the kubeflow packages into your application.
+
+```
+cd ${APP_NAME}
+# Add the kubeflow registry to the ksonnet app
+ks registry add kubeflow github.com/kubeflow/kubeflow/tree/master/kubeflow
+# Download the kubeflow/core ksonnet package
+ks pkg install kubeflow/core
+```
+
+Create the kubeflow core component. We will do all our work in the
+`issue-summarization-model` kubernetes namespace.
+
+```
+ks generate core kubeflow-core --name=kubeflow-core --namespace=${KUBE_NS}
+```
+
+ksonnet allows multiple
+[environments](https://ksonnet.io/docs/concepts#environment), but for simplicity
+we will be using the `default` environment for this tutorial.
+
+```
+KSONNET_ENV=default
+ks env set ${KSONNET_ENV} --namespace ${KUBE_NS}
+```
+
+If using GKE, we can configure our cloud environment to use GCP features with a
+single parameter:
+
+```
+ks param set kubeflow-core cloud gke --env=${KSONNET_ENV}
+```
+
+### Provision storage for training data
```

Comment:
See kubeflow/kubeflow#240 and kubeflow/kubeflow#241. Can we create docs in a central location that cover topics like how to setup storage?

@nkashy1 Were you going to take a look at doing that?